src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -473,6 +473,7 @@ public virtual Expression BindBinaryOperatorNode(BinaryOperatorNode binaryOperat
             Expression left = Bind(binaryOperatorNode.Left);
             Expression right = Bind(binaryOperatorNode.Right);
 
+            bool createDateBinaryExpression = IsDate(binaryOperatorNode);
```

Comment:
date() method is only used in binary operator?

Reply:
I think yes